lib/rspec-puppet-facts.rb

```diff
@@ -103,25 +103,34 @@ def on_supported_os(opts = {})
 
     # FacterDB may have newer versions of facter data for which it contains a subset of all possible
     # facter data (see FacterDB 0.5.2 for Facter releases 3.8 and 3.9). In this situation we need to
-    # cycle through and downgrade Facter versions per platform type until we find matching Facter data.
+    # cycle through and search Facter versions per platform type until we find matching Facter data.
+    # Up to 10 versions will be checked (5 in either direction).
     filter.each do |filter_spec|
       facter_version_filter = RspecPuppetFacts.facter_version_to_filter(facterversion)
       db = FacterDB.get_facts(filter_spec.merge({ :facterversion =>  facter_version_filter }))
 
       version = facterversion
-      while db.empty? && version !~ /\d+\.0($|\.\d+)/
-        version = RspecPuppetFacts.down_facter_version(version)
+      tries = 0
+      while db.empty? && tries < 10
```

Comment:
I saw at least a few systems in facterdb where the gap to find a matching version of Facter would take more than 5 version changes. Maybe change this to 20? If there is a "close" match it will still find it just as fast. If there isn't one it will take longer but have a better chance of success. I'd rather it take extra time and return facts than bail out without being able to find any facts.

Reply:
Updated